src/rabbit_shovel_worker.erl

```diff
@@ -227,14 +231,27 @@ publish(Tag, Method, Msg,
                         decr_remaining(1, State)
       end).
 
-make_conn_and_chan(URIs) ->
+make_conn_and_chan(URIs, ShovelName) ->
     URI = lists:nth(rand_compat:uniform(length(URIs)), URIs),
     {ok, AmqpParam} = amqp_uri:parse(URI),
-    {ok, Conn} = amqp_connection:start(AmqpParam),
+    ConnName = get_connection_name(ShovelName),
+    {ok, Conn} = amqp_connection:start(AmqpParam, ConnName),
     link(Conn),
     {ok, Chan} = amqp_connection:open_channel(Conn),
     {Conn, Chan, list_to_binary(amqp_uri:remove_credentials(URI))}.
 
+%% for static shovels, name is an atom from the configuration file
+get_connection_name(ShovelName) when is_atom(ShovelName) ->
+    list_to_binary([<<"Shovel">>, <<" ">>, atom_to_binary(ShovelName, utf8)]);
```

Comment:
You can [concatenate binaries directly](http://stackoverflow.com/questions/600642/how-do-i-concatenate-two-binaries-in-erlang)